2.4/root/usr/share/container-scripts/mongodb/common.sh

```diff
@@ -4,15 +4,23 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
-# Used for wait_for_mongo_* functions
-MAX_ATTEMPTS=60
-SLEEP_TIME=1
-
+# Data directory where MongoDB database files live. The data subdirectory is here
+# because mongodb.conf lives in /var/lib/mongodb/ and we don't want a volume to
+# override it.
+export MONGODB_DATADIR=/var/lib/mongodb/data
 export MONGODB_CONFIG_PATH=/etc/mongod.conf
-export MONGODB_PID_FILE=/var/lib/mongodb/mongodb.pid
-export MONGODB_KEYFILE_PATH=/var/lib/mongodb/keyfile
+export MONGODB_KEYFILE_PATH="${HOME}/keyfile"
 export CONTAINER_PORT=27017
 
+# Configuration settings.
+export MONGODB_NOPREALLOC=${MONGODB_NOPREALLOC:-true}
+export MONGODB_SMALLFILES=${MONGODB_SMALLFILES:-true}
+export MONGODB_QUIET=${MONGODB_QUIET:-true}
+
+# Constants used for waiting
+export MAX_ATTEMPTS=60
+export SLEEP_TIME=1
```

Comment:
Also MONGODB_CONFIG_PATH and MONGODB_KEYFILE_PATH, or not?

/usr/share/container-scripts/mongodb/common.sh is sourced in every run or execed file and config file does not contain them... so it is necessary to export them.